sdks/rust/tests/procedure-client/src/main.rs

```diff
@@ -310,5 +313,44 @@ fn exec_procedure_http_err() {
             })
         }
     });
+
+    test_counter.wait_for_all();
+}
+
+fn exec_schedule_procedure() {
+    let test_counter = TestCounter::new();
+    let sub_applied_nothing_result = test_counter.add_test("on_subscription_applied_nothing");
+
+    let mut callback_result = Some(test_counter.add_test("insert_with_tx_commit_callback"));
+
+    connect_then(&test_counter, {
+        move |ctx| {
+            ctx.db().proc_inserts_into().on_insert(move |_, row| {
+                assert_eq!(row.x, 42);
+                assert_eq!(row.y, 24);
+
+                // Ensure that the elapsed time
+                // between the reducer and procedure
+                // is at least 1 second
+                // but no more than 1 second + 50 milliseconds.
```

Comment:
That seems optimistic, given that we run these tests in CI with a debug build of the host.

Reply:
Second this, I witnessed the replication tests with even larger time margins being flaky in CI :|